### ISSUE TITLE
Avoid detecting text of type union.

### ISSUE THREAD

**Issue by junkisai** [edvardchen/eslint-plugin-i18next#114]
We would like to prevent Lint from detecting values with the union type specified, as shown below.
Currently, it is detected as an error.

```ts
type Animal = 'dog' | 'cat'

const animal: Animal = 'dog' -> disallow literal string: sample: Animal = 'dog'eslint(i18next/no-literal-string)
```

---

This PR https://github.com/edvardchen/eslint-plugin-i18next/pull/56 has been deleted in what appears to be the relevant code. Do you plan to implement similar changes again?

**Comment by edvardchen** [comment]
My original thought is to ignore literal strings that appears in the type annotations

But TypeScript will treat `const a = 'foo'` as `const a:'foo' = 'foo'` so I reverted that. 

I think we can give it a try, to ignore strings that appears in **union type**
